### ISSUE TITLE
Figure out what to do about exact alarm API restriction

### ISSUE THREAD

**Issue by mhoran** [ubergeek42/weechat-android#545]
Google Play is going to be removing the ability for apps to use the exact alarm API unless they are alarm clock or calendar apps. We use this for ping functionality, and inexact alarms aren't very useful to keep connections alive.

We can either remove from the Play store (rely on F-Droid or hosted APKs to install) or potentially remove the permission and require that users disable batter optimizations for the app, which supposedly allows use of this "restricted" API.

Other apps (e.g. K-9 Mail, FairEmail) take the latter approach (requiring battery optimization disable.) They bring users to the battery optimization page when starting up the app and optimizations are enabled.

All this is a bunch of annoying extra work we have to do thanks to Google. I'm inclined to just remove from the app store, but we still have ~800 users there so maybe we can do this extra work, sometime.

**Comment by oakkitten** [comment]
I think we [can ask](https://developer.android.com/reference/android/Manifest.permission#SCHEDULE_EXACT_ALARM:~:text=Apps%20need%20to%20target%20API%20Build.VERSION_CODES.S%20or%20above%20to%20be%20able%20to%20request%20this%20permission.%20Note%20that%20apps%20targeting%20lower%20API%20levels%20do%20not%20need%20this%20permission%20to%20use%20exact%20alarm%20APIs.) for [alarm permissions](https://developer.android.com/about/versions/13/features#use-exact-alarm-permission:~:text=If%20your%20app%20sets%20exact%20alarms%20but%20doesn%27t%20satisfy%20either%20case%20shown%20in%20the%20previous%20list%2C%20continue%20to%20declare%20the%20SCHEDULE_EXACT_ALARM%20permission%20instead%2C%20and%20be%20prepared%20for%20the%20situation%20where%20the%20user%20denies%20access%20to%20your%20app.) now?

**Comment by oakkitten** [comment]
Got this crash from from an user just today, app installed from F-Droid

```
I 🐶    : r-1  : Handshake WeeChat version: 0x3070100
E AndroidRuntime: FATAL EXCEPTION: e-1
E AndroidRuntime: Process: com.ubergeek42.WeechatAndroid, PID: 7894
E AndroidRuntime: java.lang.SecurityException: Caller com.ubergeek42.WeechatAndroid needs to hold android.permission.SCHEDULE_EXACT_ALARM to set exact alarms.
E AndroidRuntime:    at android.os.Parcel.createExceptionOrNull(Parcel.java:2426)
E AndroidRuntime:    at android.os.Parcel.createException(Parcel.java:2410)
E AndroidRuntime:    at android.os.Parcel.readException(Parcel.java:2393)
E AndroidRuntime:    at android.os.Parcel.readException(Parcel.java:2335)
E AndroidRuntime:    at android.app.IAlarmManager$Stub$Proxy.set(IAlarmManager.java:359)
E AndroidRuntime:    at android.app.AlarmManager.setImpl(AlarmManager.java:947)
E AndroidRuntime:    at android.app.AlarmManager.setImpl(AlarmManager.java:907)
E AndroidRuntime:    at android.app.AlarmManager.setExact(AlarmManager.java:705)
E AndroidRuntime:    at com.ubergeek42.WeechatAndroid.service.RelayService.hello(SourceFile:42)
E AndroidRuntime:    at com.ubergeek42.WeechatAndroid.service.RelayService.onStateChanged(SourceFile:111)
E AndroidRuntime:    at com.ubergeek42.weechat.relay.connection.RelayConnection$$ExternalSyntheticLambda4.run(SourceFile:134)
E AndroidRuntime:    at androidx.core.app.ActivityCompat$$ExternalSyntheticLambda0.run(SourceFile:538)
E AndroidRuntime:    at com.ubergeek42.weechat.relay.connection.Utils$FriendlyThread.run(Unknown Source:7)
E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
E AndroidRuntime:    at com.android.server.alarm.AlarmManagerService$5.set(AlarmManagerService.java:2547)
E AndroidRuntime:    at android.app.IAlarmManager$Stub.onTransact(IAlarmManager.java:194)
E AndroidRuntime:    at android.os.Binder.execTransactInternal(Binder.java:1179)
E AndroidRuntime:    at android.os.Binder.execTransact(Binder.java:1143)
```

[See also](https://developer.android.com/training/scheduling/alarms#exact-permission-declare)

**Comment by oakkitten** [comment]
Solved in 0558c22, or rather 3b434688df2202c65bc25cf7d611c89426c1415a
